Pcap++/src/PcapLiveDeviceList.cpp

```diff
@@ -39,32 +41,30 @@ PcapLiveDeviceList::~PcapLiveDeviceList()
 
 void PcapLiveDeviceList::init()
 {
-	pcap_if_t* interfaceList;
-	char errbuf[PCAP_ERRBUF_SIZE];
-	int err = pcap_findalldevs(&interfaceList, errbuf);
-	if (err < 0)
+	std::unique_ptr<pcap_if_t, internal::PcapFreeAllDevsDeleter> interfaceList;
+	try
+	{
+		interfaceList = internal::getAllLocalPcapDevices();
+	}
+	catch (const std::exception& e)
 	{
-		PCPP_LOG_ERROR("Error searching for devices: " << errbuf);
+		PCPP_LOG_ERROR(e.what());
```

Comment:
recommend to have a meaningful error message here.

Reply:
The error messages are actually the same as before. It is just that the DeviceUtil functions have the error messages thrown in a `runtime_error` as the exception reason and this code catches that and logs the message in the exception.